src/host/ut_host/ScreenBufferTests.cpp

```diff
@@ -2502,6 +2506,108 @@ void ScreenBufferTests::SetColorTableThreeDigits()
 
 }
 
+
+void ScreenBufferTests::SetDefaultForegroundColor()
+{
+    // Setting the default foreground color should work
+
+    CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    gci.LockConsole(); // Lock must be taken to swap buffers.
+    auto unlock = wil::scope_exit([&] { gci.UnlockConsole(); });
+
+    SCREEN_INFORMATION& mainBuffer = gci.GetActiveOutputBuffer();
+    VERIFY_IS_FALSE(mainBuffer._IsAltBuffer());
+    WI_SetFlag(mainBuffer.OutputMode, ENABLE_VIRTUAL_TERMINAL_PROCESSING);
+    VERIFY_IS_TRUE(WI_IsFlagSet(mainBuffer.OutputMode, ENABLE_VIRTUAL_TERMINAL_PROCESSING));
+
+    StateMachine& stateMachine = mainBuffer.GetStateMachine();
+
+    COLORREF originalColor = gci.GetDefaultForegroundColor();
+    COLORREF newColor = gci.GetDefaultForegroundColor();
+    COLORREF testColor = RGB(0x33, 0x66, 0x99);
+    VERIFY_ARE_NOT_EQUAL(originalColor, testColor);
+
+    // Valid hexadecimal notation
+    std::wstring seq = L"\x1b]10;rgb:33/66/99\x1b\\";

```

Comment:
To me, the exciting part about hex notation is using A-F somewhere. I'm concerned using only 0-9 digits misses the opportunity to guarantee that it is actually parsing hex-wise and not decimal-wise

Reply:
Sure 😀 -- My concern is basically just getting the 10 and 11 part right, because the rest of it is just using parsing that already existed, but I'll add another example here.